qiskit_research/utils/pulse_scaling.py

```diff
@@ -94,7 +95,28 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
                     if len(chunk) > 1:
                         for node in chunk[1:]:
                             dag.remove_op_node(node)
-            return dag
+        return dag
+
+
+class ReduceAngles(TransformationPass):
+    """Reduce angle of scaled pulses to between -pi and pi.
+
+    This works only after Parameters are bound. Gate strings
+    should only be single-parameter scaled pulses, i.e.
+    'rzx' and 'secr'.
+    """
+
+    def __init__(self, gate_strs: List[str]):
```

Comment:
- rename `gate_strs -> gate_names`
- use `list[str] instead of List[str]`
- add a docstring where you documenting the argument. Say that it assumes each named gate has a single real-valued parameter.

Reply:
Done!